CSharp/README.md

```diff
@@ -23,6 +24,115 @@ var httpClient = new HttpClient();
 IUserService userService = new UserService();
 ```
 
+### Don't use explicit *this* reference
+
+It is redundant and does not add for readability. For more information read [this SO post](http://stackoverflow.com/questions/5885249/to-use-or-not-to-use-the-this-qualifier-in-c-sharp).
+
+```csharp
+// Bad
+this.ValidateParameters();
+
+// Good
+ValidateParameters();
+```
+
+### Use *string* instead of *String* keyword

```

Comment:
Thank you all. I will do this improvement but not sure about doing the .editorconfig, .vssetting and .ruleset thing now. I do believe we should add them but in the worst scenario we can add those later, once we define the rules.